### PR TITLE
regex warning

### PR DESCRIPTION
**Pull request by bounceme**

### CODE REVIEW

**Comment by bounceme** [comment]
https://groups.google.com/forum/#!searchin/vim_dev/7.4.002/vim_dev/GfSwrDEAejM/zqgZJRHo1AEJ
this is the patch which fixes the issue with highlighting right? Also regexpengine is a global option

File: ftplugin/javascript.vim [hunk 1]
```diff
@@ -1 +1,4 @@
 setlocal suffixesadd+=.js
+if v:version > 704 || (v:version == 704 && has('patch002'))
```
Comment (amadeus) [inline ftplugin/javascript.vim]: Wait, this says if versions is GREATER than 704, shouldn't it LESS THAN?

File: ftplugin/javascript.vim [hunk 2]
```diff
@@ -1,4 +1,4 @@
 setlocal suffixesadd+=.js
-if v:version > 704 || (v:version == 704 && has('patch002'))
+if (v:version < 704 || (v:version == 704 && has('patch002'))) && exists('&regexpengine')
```
Comment (amadeus) [inline ftplugin/javascript.vim]: We need a NOT on the `has()` patch, since patch fixes, and therefore, we don't want `re=1` on `has()` being truthy.

**Comment by bounceme** [comment]
how about now?

**Comment by amadeus** [comment]
Looks good to me. Can you squash these commits locally though, and give them a better name before merging?

**Comment by bounceme** [comment]
sorry,the pr name should be changed? you can squash and edit/remove the commit messages when merging on github now

**Pull request closed without merging**